### PR TITLE
Add plugin: Foliate

### PR DESCRIPTION
**Pull request by xhuajin**
# I am submitting a new Community Plugin

- [x] I attest that I have done my best to deliver a high-quality plugin, am proud of the code I have written, and would recommend it to others. I commit to maintaining the plugin and being responsive to bug reports. If I am no longer able to maintain it, I will make reasonable efforts to find a successor maintainer or withdraw the plugin from the directory.

## Repo URL

Link to my plugin: https://github.com/xhuajin/foliate

## Release Checklist

- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugin's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by Zachatoo** [comment]
[# Obsidian Foliate](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

Your repo mentions a "foliate-js" package that is defined in your packages folder, but it is not checked in to source control. If that code is closed source, then please provide me with access to that code. Otherwise please include it in your repo so it can be reviewed. Or you can reference an npm package and I can validate against that.

[styles.css](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/styles.css)
Please don't include generated files in your repo.

[--radius: 0.625rem;](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L38-L70), [--border: hsl(217.2 32.6% 17.5%);](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L75-L93), [.empty-state {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L352), [.empty-icon {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L357), [.progress-text {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L461), [.progress-bar {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L467), [.progress-fill {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L476), [.recent-badge {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L496), [.stat-card {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L517), [.stat-number {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L525), [.stat-label {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L533), [.context-menu {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L629), [.progress-animated {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles/coverflow.css#L50), [[data-slot='tree-item-label']::before {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles/epub-toc.css#L22)
Please update your classes to be more specific to your plugin to avoid conflicts with other plugins.

[const tocView = existingLeaf.view as EpubTocView;](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/main.ts#L289), [const view = leaf.view as EpubView;](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/main.ts#L222), [const view = leaf.view as EpubView;](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/view/FoliateView.tsx#L527)
Don't cast using `as` for a `View`, validating using an `instanceof` check instead.

[import moment from 'moment';](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/view/FoliateView.tsx#L36)
You should import this from `"obsidian"` instead.

['Open Reading History'](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/lang/locale/en.ts#L44), ['Reading History'](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/lang/locale/en.ts#L84), ['My Reading History'](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/lang/locale/en.ts#L85), ['Download Image'](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/lang/locale/en.ts#L165), ['Copy to Clipboard'](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/lang/locale/en.ts#L166), ['This will delete ALL reading records. Continue?'](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/lang/locale/en.ts#L151), ['Remove ALL stored reading progress (irreversible)'](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/lang/locale/en.ts#L148)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI), product / brand names follow the casing set by the respective owner.

[const styleEl = document.createElement('style');](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/view/FoliateView.tsx#L1255)
You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.

[existingLeaf.detach();](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/view/FoliateView.tsx#L533)
Don't detach your custom view to refresh it, your custom view may end up moving elsewhere in the workspace when you do this.

[if (content.startsWith('---')) {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/view/FoliateView.tsx#L602-L619)
Use [`getFrontMatterInfo`](https://docs.obsidian.md/Reference/TypeScript+API/getFrontMatterInfo) to get information about the frontmatter of a file, including whether there is a frontmatter block, the offsets of where it starts and ends, and the frontmatter text.

[if (!(await app.vault.adapter.exists(noteFolderPath))) {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/view/FoliateView.tsx#L894)
[Prefer the Vault API over the Adapter API](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Prefer+the+Vault+API+over+the+Adapter+API)

---

[* {](https://github.com/xhuajin/foliate/blob/961cc2cb893e61e996ae12082c3fbc46d4c838fd/src/styles.css#L612-L617)
Please don't overwrite core styling in a plugin, add another class/data attribute so it only applies to your plugin.

---

Please note that this is an initial review, a more thorough review will occur when the above changes have been made. The bot will mark this PR as "changes made" when it has detected that you have made all the required changes. The bot checks about every 6 hours.
